Simplified/Adobe DRM/AdobeDRMContainer.mm

```diff
@@ -0,0 +1,156 @@
+        
+        ePub3::ErrorHandlerFn sdkErrorHandler = ^(const ePub3::error_details& err) {
+            BOOL isSevereEpubError = NO;
+            if (err.is_spec_error()
+                    && (err.severity() == ePub3::ViolationSeverity::Critical
+                    || err.severity() == ePub3::ViolationSeverity::Major))
+                isSevereEpubError = YES;
+
+            return ePub3::DefaultErrorHandler(err);
+        };
+        ePub3::SetErrorHandler(sdkErrorHandler);
+
+        ePub3::InitializeSdk();
+        ePub3::PopulateFilterManager();
+        ePub3::AdeptFilter::Register();
+
+        try {
+            container = ePub3::Container::OpenContainer(path.UTF8String);
+            
+        }
+        catch (std::exception& e) { // includes ePub3::ContentModuleException
+            auto msg = e.what();
+            std::cout << msg << std::endl;
+        }
+        catch (...) {
+        }
+        
+        if (container == nullptr) {
+            return nil;
+        }
+
+        package = container->DefaultPackage().get();
+        ePub3::string s = package->TableOfContents()->SourceHref();
+        manifestItem = package->ManifestItemAtRelativePath(s);
+}
+    return self;
+}
+
+- (void *)getDecodedByteStream:(void *)currentByteStream isRangeRequest:(BOOL)isRangeRequest {
+        // Get the manifest item initWithURL: saves for further use
+        ePub3::ManifestItemPtr m = std::const_pointer_cast<ePub3::ManifestItem>(manifestItem);
+        size_t numFilters = package->GetFilterChainSize(m);
+        ePub3::ByteStream *byteStream = nullptr;
+        ePub3::SeekableByteStream *rawInput = (ePub3::SeekableByteStream *)currentByteStream;
+        
+        if (numFilters == 0)
+        {
+            byteStream = (ePub3::ByteStream *) currentByteStream; // is actually a SeekableByteStream
+        }
+        else if (numFilters == 1 && isRangeRequest)
+        {
+            byteStream = package->GetFilterChainByteStreamRange(m, rawInput).release(); // is *not* a SeekableByteStream, but wraps one
+            if (byteStream == nullptr)
+            {
+                byteStream = package->GetFilterChainByteStream(m, rawInput).release(); // is *not* a SeekableByteStream, but wraps one
+            }
+        }
+        else
+        {
+            byteStream = package->GetFilterChainByteStream(m, rawInput).release(); // is *not* a SeekableByteStream, but wraps one
+        }
+        
+        return byteStream;
+
+}
+
+
+- (NSData *)decodeData:(NSData *)data {
+    
+    @synchronized (acsdrm_lock) {
+
+        NSData *m_data;
+        NSUInteger m_contentLength;
+        NSUInteger m_contentLengthCheck;
+        UInt8 m_buffer[1024 * 256];
+        std::unique_ptr<ePub3::ByteStream> m_byteStream;
+        DataByteStream *dataByteStream;
+        
+        dataByteStream = new DataByteStream((unsigned char *)[data bytes], (unsigned long)data.length);
+        
+        if (m_data == nil) {
```

Comment:
I'm probably missing something, but isn't this just a local variable declared at 114? But that var is not initialized there... maybe you meant to initialize `m_data` with `data`?

Reply:
a good catch! no need for m_data here, got rid of a function that was reading the stream and returning result in m_data, but didn't remove the variable